src/backend/commands/createas.c

```diff
@@ -326,6 +325,12 @@ ExecCreateTableAs(CreateTableAsStmt *stmt, const char *queryString,
 		save_nestlevel = NewGUCNestLevel();
 	}
 
+	copied_query = copyObject(query);
+	if (is_matview && into->ivm)
+	{
+		copied_query = rewriteQueryForIMMV(copied_query, into->colNames, false);
+	}
+
 	if (into->skipData)
```

Comment:
> 
> 
>     * {,} is not necessary since there is only one line in if block.
Ok, I fix it.

>     * copyObject seems unnecessary here for IMMV cases because this is done in rewriteQueryForIMMV.
You have a point, I fix it.